python_ta/debug/accumulation_table.py

```diff
@@ -64,50 +76,73 @@ class AccumulationTable:
+    _accumulator_names: Union[list[str], list[list[str]]]
+    """Either a list of accumulator variable names to track across all loops,
+    or a list of lists where each inner list contains the accumulator names for the corresponding loop."""
+    loops: list[dict[str, Any]]
+    """A list of dictionaries containing loop-specific data for each loop in the with block."""
     output_filepath: Optional[str]
     output_format: str
 
     def __init__(
         self,
-        accumulation_names: list[str],
+        accumulation_names: Union[list[str], list[list[str]]],
         output: Union[None, str] = None,
         format: Literal["table", "csv"] = "table",
     ) -> None:
         """Initialize an AccumulationTable context manager for print-based loop debugging.
 
         Args:
-            accumulation_names: a list of the loop accumulator variable names to display.
-
+            accumulation_names: either a list of accumulator variable names to track across all loops,
+                or a list of lists where each inner list contains the accumulator names for the corresponding loop.
+            output: optional filepath where the table will be written. If None, prints to stdout.
+            format: output format, either "table" (formatted table) or "csv" (CSV format).
         """
-        self.loop_accumulators = {accumulator: [] for accumulator in accumulation_names}
-        self.loop_variables = {}
-        self._loop_lineno = 0
+        self._accumulator_names = accumulation_names
+        self.loops = []
         self.output_filepath = output
         self.output_format = format
 
-    def _record_iteration(self, frame: types.FrameType) -> None:
-        """Record the values of the accumulator variables and loop variables of an iteration"""
-        if self.loop_variables != {} and len(list(self.loop_variables.values())[0]) > 0:
-            for loop_var in self.loop_variables:
-                self.loop_variables[loop_var].append(copy.deepcopy(frame.f_locals[loop_var]))
+    @property
+    def loop_accumulators(self) -> dict[str, list]:
+        """Read-only access to accumulator values. Only works with a single loop in context manager."""
+        if len(self.loops) == 1:
+            return self.loops[0]["loop_accumulators"]
+        raise ValueError("Only available when tracking a single loop")
+
+    @property
+    def loop_variables(self) -> dict[str, list]:
+        """Read-only access to loop variable values. Only works with a single loop in context manager."""
+        if len(self.loops) == 1:
+            return self.loops[0]["loop_variables"]
+        raise ValueError("Only available when tracking a single loop")
+
+    def _record_iteration(self, frame: types.FrameType, lst_index: int) -> None:
+        """Record the values of the accumulator variables and loop variables of an iteration for a given loop at index
+        lst_index."""
+        if (
+            self.loops[lst_index]["loop_variables"] != {}
```

Comment:
In this and the other functions there's a fair amount of duplicated code with nested attribute/dict access. I would define variables right at the very start for `loop_variables = self.loops[lst_index]["loop_variables"]` (and for `loop_accumulators`) and then just use these variables throughout. That'll help make the code look a bit cleaner.